pygmt/tests/test_figure.py

```diff
@@ -292,6 +292,27 @@ def mock_psconvert(*args, **kwargs):  # pylint: disable=unused-argument
     }
 
 
+def test_figure_savefig_worldfile():
+    """
+    Check if a world file is created for supported formats and raise an error
+    for unsupported formats.
+    """
+    fig = Figure()
+    fig.basemap(region=[0, 1, 0, 1], projection="X1c/1c", frame=True)
+    # supported formats
+    for fmt in [".bmp", ".jpg", ".png", ".tif"]:
```

Comment:
Need to add `.ppm` if #2771 is merged:
```suggestion
    for fmt in [".bmp", ".jpg", ".png", ".ppm", ".tif"]:
```